sources/web/datalab/server.ts

```diff
@@ -275,7 +275,7 @@ function uncheckedRequestHandler(request: http.ServerRequest, response: http.Ser
       urlpath.indexOf('/oauthcallback') == 0) {
     // Start or return from auth flow.
     auth.handleAuthFlow(request, response, parsed_url, appSettings);
-  } else if (isStaticResource(urlpath)) {
+  } else if (!reverseProxy.isReverseProxyRequest(request) && isStaticResource(urlpath)) {
```

Comment:
While I think this code is OK for now, I do find the request handling code a bit tedious to follow. We have requestHandler() that calls uncheckedRequestHandler() that calls handleRequest(), which doesn't provide much distinction. The comment for uncheckedRequestHandler implies there are two ways we process requests, and the method then has two branches, but I'm not convinced that those two branches actually correspond to the two ways to process requests implied in the comment. It might be good to do a bit of refactoring or renaming to clarify, or at least so that the function comment is more closely aligned with the names of the functions that it is calling.